Sources/Shapes.swift

```diff
@@ -291,6 +302,23 @@ public extension Mesh {
         )
     }
 
+    static func ellipticCylinder(width: Double = 1,
+                                 depth: Double = 2,
+                                 height: Double = 1,
+                                 segments: Int = 16,
+                                 poleDetail: Int = 0,
```

Comment:
`poleDetail` isn't used here, since it's a feature of the lathe method of construction. Now that I think about it, it's a bit weird that `cylinder` uses lathe and `ellipticCylinder` uses extrude (although I understand why, obviously). I wonder if it might be better to leave `ellipticCylinder` out of the stdlib for now.

Reply:
oh sorry, I was offline, didn't have time to clean this up. I can see you merged this as is, let me  know  if there's a need in a follow up to clean this up or to refactor elliptical cylinder into external library

Reply:
I merged it, but then removed the ellipticCylinder in a subsequent commit. 

I think extruding an ellipse is probably simple enough that it doesn't need to be built-in primitive, and the fact that the implementation uses extrude where cyclinder uses lathe us unexpected (it makes a big difference for how the texture coordinates are generated).

Note that you can also create an elliptic cylinder very easily by applying a nonuniform scale to a regular cylinder.